apps/src/blockly/addons/cdoUtils.ts

```diff
@@ -58,6 +59,48 @@ export function loadBlocksToWorkspace(
   Blockly.serialization.workspaces.load(mainSource, workspace);
   positionBlocksOnWorkspace(workspace);
   Blockly.hasLoadedBlocks = true;
+
+  // Dynamically add procedure call blocks to an uncategorized toolbox
+  // if specified in the level config (e.g. Minecraft Agent levels).
+  // Levels will include: "top_level_procedure_autopopulate": "true"
+  if (Blockly.topLevelProcedureAutopopulate) {
+    addProcedureCallBlocksToFlyout(workspace, mainSource);
+  }
+}
+
+function addProcedureCallBlocksToFlyout(
+  workspace: WorkspaceSvg,
+  mainSource: WorkspaceSerialization
+) {
+  // options.languageTree is the translated toolbox info
+  if (workspace.getFlyout() && workspace.options?.languageTree) {
+    const callBlocks = [] as ToolboxItemInfo[];
+    const defBlocks = mainSource.blocks.blocks.filter(
```

Comment:
Nit, non-blocking: Prefer just to write out definition blocks here

Reply:
Agreed. If there are no other changes requested, I may save this for an immediate follow-up so we don't have to rerun drone.

Reply:
Agreed!